relays/lib-substrate-relay/src/lib.rs

```diff
@@ -96,3 +98,45 @@ impl<AccountId> TaggedAccount<AccountId> {
 		}
 	}
 }
+
+/// Batch call builder.
+pub trait BatchCallBuilder<Call> {
+	/// If `true`, then batch calls are supported at the chain.
+	const BATCH_CALL_SUPPORTED: bool;
+
+	/// Create batch call from given calls vector.
+	fn build_batch_call(_calls: Vec<Call>) -> Call;
+}
+
+impl<Call> BatchCallBuilder<Call> for () {
+	const BATCH_CALL_SUPPORTED: bool = false;
+
+	fn build_batch_call(_calls: Vec<Call>) -> Call {
+		unreachable!(
```

Comment:
Yeah, it is better to be safe here :)

Reply:
Done

Reply:
TBH I'm still not sure about `BATCH_CALL_SUPPORTED`. We're still relying on the fact that if `BATCH_CALL_SUPPORTED` is false, no one will call `build_batch_call`. Otherwise we're getting a runtime error. Ideally we would have a mechanism that checks this at compile time, but I don't know if it's possible.

From what I understand `BATCH_CALL_SUPPORTED` is a way of saying "This end of the lane doesn't support batching". It would be nice if instead of this we could have separate traits. Something like:

```
trait SubstrateMessageLaneWithBatchingAtSource {
	type BatchCallBuilder: BatchCallBuilder;
    ....
}
```

```
trait SubstrateMessageLaneWithBatchingAtTarget {
    ...
}
```

Such that when the Source or the destination doesn't support batching, we don't need to use `()` and we just don't implement the trait. But then I'm not sure if this would be possible at all, and how much it would complicate the current logic in the callers.

Anyway, I didn't have time to think about this too much. It might not make sense at all. And it's not a blocker. But just saying that I would like to experiment with this idea, or something similar in the future when I have some spare time and maybe come up with a PR if I can get anything to work.

Reply:
Yeah - I agree that it isn't perfect and would be better to define it at compile time. I'll prepare an issue for that

Reply:
#1724 1724